src/Test/Plutip/Contract.hs

```diff
@@ -305,36 +317,41 @@ withContractAs walletIdx toContract = do
       it is important to preserve this order for Values check with `assertValues`
       as there is no other mechanism atm to match `TestWallet` with collected `Value`
       -}
+      collectValuesPkhs :: NonEmpty PaymentPubKeyHash
       collectValuesPkhs = fmap ledgerPaymentPkh wallets'
 
-      -- wallelt `PaymentPubKeyHash`es that will be available in
+      -- wallet `PaymentPubKeyHash`es that will be available in
       -- `withContract` and `withContractAs`
+      otherWalletsPkhs :: [PaymentPubKeyHash]
       otherWalletsPkhs = fmap ledgerPaymentPkh otherWallets
-      contract =
-        wrapContract
-          collectValuesPkhs
-          (toContract otherWalletsPkhs)
-  liftIO $ runContract cEnv ownWallet contract
+
+  -- run the test contract
+  execRes <- liftIO $ runContract cEnv ownWallet (toContract otherWalletsPkhs)
+
+  -- run the 'wrapContract' to get all the values present at the test wallets.
+  execValues <- liftIO $ runContract cEnv ownWallet (wrapContract @w @s @e collectValuesPkhs)
```

Comment:
Looking at current code layout I came to thinking -  do we even need `wrapContract`? It is used only here. Maybe something like this will be more straightforward and easier to read/parse mentally:

```haskell
execRes <- liftIO $ runContract cEnv ownWallet (toContract otherWalletsPkhs)

void (waitNSlots 1) -- wait for contract to finish

execValues <- collectValuesFromWallets phksToCollectValueFrom
```

or maybe rename `wrapContract` to something mentioning collecting Values as its logic is shrieked down to collecting values now.

What do you think?

Reply:
Fixed, by removing `wrapContract` with a local let function `valuesAtWallet`.